### ISSUE TITLE
Compliance/Earned Credits - External EC request status (pending/approved)

### ISSUE THREAD

**Issue by NicoleGovvy** [bcgov/cas-compliance#60]
#### Description:
An operator can view the status of their EC issuance request in Compliance.

#### Acceptance Criteria:
As an operator, given that I've submitted an EC issuance request for an operation under the emissions threshold, when I navigate to the 'Track Status of Issuance' page, then I can see the status of my request.

Figma design for [pending request](https://www.figma.com/proto/LsDmLDCdnJqI8UahyqvOD8/BCIERS-Designs?node-id=21296-1319&t=0lZgimEioqAnwSTV-1)
Figma design for [approved request](https://www.figma.com/proto/LsDmLDCdnJqI8UahyqvOD8/BCIERS-Designs?node-id=21296-1412&t=2toId8lC864I05SZ-1)

Display: 

- [ ] When the request is pending: Information about what will happened when the request is approved. 
- [ ] When the request is approved: a success message saying the credits have been deposited to the designed BCCR account
- [ ] Number of credits requested
- [ ] Date request submitted
- [ ] Status of issuance
- [ ] Date request resolved (approved only)
- [ ] BCCR Trading name
- [ ] BCCR holding account number

#### Development Checklist:

- [ ] Create `bc_obps/compliance/models/issuance_request.py` model
- [ ] Create a new endpoint in `bc_obps/compliance/api/issuance_api.py`:
  - [ ] GET endpoint to retrieve issuance status by compliance summary ID
- [ ] Create `bciers/apps/compliance/src/app/bceidbusiness/industry_user_admin/compliance-summaries/[compliance_summary_id]/track-issuance/page.tsx`
- [ ] Implement the page layout based on the Figma designs for both states:
  - [ ] Pending request state with information about what will happen when approved
  - [ ] Approved request state with success message about credits deposited
- [ ] Create a component to display different status messages based on the request status
- [ ] Implement appropriate styling for each status type:
  - [ ] Yellow/orange for pending
  - [ ] Green for approved
- [ ] Add a link to the Track Status page in the compliance summary navigation
- [ ] Make the link conditionally visible based on whether a request has been submitted
- [ ] Test the page rendering for different status values (unit tests)
- [ ] Test that the correct information is displayed based on status (unit tests)
- [ ] Test navigation and routing (unit tests)

#### Definition of Ready (Note: If any of these points are not applicable, mark N/A)

- [ ] User story is included
- [ ] User role and type are identified
- [ ] Acceptance criteria are included
- [ ] Wireframes are included (if required)
- [ ] Design / Solution is accepted by Product Owner
- [ ] Dependencies are identified (technical, business, regulatory/policy)
- [ ] Story has been estimated (under 13 pts)

#### Definition of Done (Note: If any of these points are not applicable, mark N/A)

- [ ] Acceptance criteria are tested by the CI pipeline
- [ ] UI meets accessibility requirements
- [ ] Configuration changes are documented, documentation and designs are updated
- [ ] Passes code peer-review
- [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
- [ ] Ticket is ready to be merged to main branch
- [ ] Can be demoed in Sprint Review
- [ ] Bugs or future work cards are identified and created
- [ ] Reviewed and approved by Product Owner

#### Blocked By/Blocking
Blocked by #59

**Comment by NicoleGovvy** [comment]
@nanyangpro is there a list of statuses for EC issuance requests? I'd written down approved, denied, and pending in the sticky note for the user story but the Figma designs have more detailed wording.

**Comment by NicoleGovvy** [comment]
@nanyangpro I've added the date the request submitted and the date the request resolved as data points to display, I can't remember if there was a reason not to include them. Happy to delete if that's the case.

**Comment by nanyangpro** [comment]
Good questions, @NicoleGovvy!
- No, we don't have a confirmed list of status. But you are right, the ones we intend for the design are:
    - Issuance not requested
    - Issuance requested, awaiting approval
    - Issuance approved
    - Issuance denied
    - Credits issued in BCCR
- Good call. We won't display the dates in the Compliance Summaries grid, but I'll add them to the 'Track Status of Issuance' page. Thanks!

**Comment by vesselak** [comment]
@nataliepiguliak please correct the operation name to appear correctly Picture
cc @fviduya

**Comment by fviduya** [comment]
I have tested this in the DEV environment, [here is the test document for reference](https://buttoninc-my.sharepoint.com/:w:/g/personal/francesca_button_is/EYpPzWu0h7ZAiP9AQjZ2leMBvcJXiPOHmxXEU6sp8zpYHA?e=UPEmhg&nav=eyJoIjoiNDEzOTU3MDQ1In0).

I can verify that:
- [x] As an operator, given that I've submitted an EC issuance request for an operation under the emissions threshold, when I navigate to the 'Track Status of Issuance' page, then I can see the status of my request.

This is currently showing just the pending status since the values are hardcoded. Moving to Pending PO for the review of test. Thank you!

**Comment by vesselak** [comment]
Reviewed QA documentation and accepting ticket.
